client/cli/src/commands/inspect_node_key.rs

```diff
@@ -17,39 +17,67 @@
 
 //! Implementation of the `inspect-node-key` subcommand
 
-use crate::{Error, NetworkSchemeFlag};
+use crate::Error;
 use clap::Parser;
 use libp2p::identity::{ed25519, PublicKey};
-use std::{fs, path::PathBuf};
+use std::{
+	fs,
+	io::{self, Read},
+	path::PathBuf,
+};
 
 /// The `inspect-node-key` command
 #[derive(Debug, Parser)]
 #[clap(
 	name = "inspect-node-key",
-	about = "Print the peer ID corresponding to the node key in the given file."
+	about = "Load a node key from a file or stdin and print the corresponding peer-id."
 )]
 pub struct InspectNodeKeyCmd {
 	/// Name of file to read the secret key from.
+	///
+	/// If not given, the secret key is read from stdin (up to EOF).
 	#[clap(long)]
-	file: PathBuf,
+	file: Option<PathBuf>,
 
-	#[allow(missing_docs)]
-	#[clap(flatten)]
-	pub network_scheme: NetworkSchemeFlag,
+	/// The input is in raw binary format.
+	///
+	/// If not given, the input is read as an hex encoded string.
+	#[clap(long)]
+	bin: bool,
 }
 
 impl InspectNodeKeyCmd {
 	/// runs the command
 	pub fn run(&self) -> Result<(), Error> {
-		let mut file_content =
-			hex::decode(fs::read(&self.file)?).map_err(|_| "failed to decode secret as hex")?;
+		let mut file_data = match &self.file {
+			Some(file) => fs::read(&file)?,
+			None => {
+				let mut buf = Vec::with_capacity(64);
+				io::stdin().lock().read_to_end(&mut buf)?;
+				buf
+			},
+		};
+
+		if !self.bin {
+			// With hex input, give to the user a bit of tolerance about whitespaces
```

Comment:
Read contents to String and call https://doc.rust-lang.org/stable/std/primitive.str.html#method.trim